src/EllCrv/Models.jl

```diff
@@ -0,0 +1,278 @@
+if $\char K = 2$ and $j(E) = 0$.
+Returns equation of the form $y^2 = x^3 + a2x^2 + a6$ 
+if $\char K = 3$ and $j(E) \neq 0$,
+Returns equation of the form $y^2 = x^3 + a4x + a6$ 
+if $\char K = 3$ and $j(E) = 0$.
+"""
+#Magma returns minimal model if base field is QQ. Not sure if we want the same.
+function simplified_model(E::EllCrv) 
+  K = base_field(E)
+  a1, a2, a3, a4, a6 = a_invars(E)
+  if characteristic(K)==2 
+    if j_invariant(E) == 0
+      return transform_rstu(E, [a2, 0, 0, 1])
+    else
+      return transform_rstu(E, [a3//a1, 0, a4//a1 + a3^2//a1^3, a1])
+    end
+  end
+  
+  if characteristic(K)==3
+    if j_invariant(E) == 0
+      return transform_rstu(E, [0, a1, a3, 1])
+    else
+      b2, b4 = b_invars(E)
+      return transform_rstu(E, [-b4//b2, a1, a3 - a1*b4//b2, 1])
+    end
+  end
+  
+  b2, b4 = b_invars(E)
+  
+  F, phi1 = transform_rstu(E, [0, -a1//2, -a3//2, 1])
+  G, phi2 = transform_rstu(F, [-b2//12, 0, 0, 1])
+  phi = phi1 * phi2
+  return G, phi, inv(phi)
+end
+
+
+@doc Markdown.doc"""
+    is_simplified_model(E::EllCrv) -> Bool
+
+Return true if E is a simplified model. 
+"""
+function is_simplified_model(E::EllCrv)
+
+  if characteristic(K)==2 || characteristic(K)==3
+    if j_invariant(E) == 0
+      return (a1, a2) == (0, 0)
+    else
+      return return (a1, a3, a4) == (1, 0, 0)
+    end
+  end
+  
+  if characteristic(K)==3
+    if j_invariant(E) == 0
+      return (a1, a2, a3) == (0, 0, 0)
+    else
+      return return (a1, a3, a4) == (0, 0, 0)
+    end
+  end
+  
+  return is_short_weierstrass_model(E)
+end
+
+
+################################################################################
+#
+#  Integral models
+#
+################################################################################
+
+@doc Markdown.doc"""
+    integral_model(E::EllCrv{fmpq}) -> (F::EllCrv{fmpz}, function, function)
+
+Given an elliptic curve $E$ over $\mathbf Q$ in short form, returns an
+isomorphic curve $F$ with model over $\mathbf Z$. The second and third
+return values are the isomorpisms $E \to F$ and $F \to E$.
+"""
+function integral_model2(E::EllCrv{fmpq})
+  _, _, _, A, B = a_invars(E)
+
+  mue = lcm(denominator(A), denominator(B))
+  Anew = mue^4 * A
+  Bnew = mue^6 * B
+  E_int = EllipticCurve([Anew, Bnew])
+
+  trafo_int = function(P) # transformes a point on E into a point on E_int
+    if P.isinfinite
+      return infinity(E_int)
+    end
+
+    xnew = mue^2 * P.coordx
+    ynew = mue^3 * P.coordy
+    Q = E_int([xnew, ynew])
+    return Q
+  end
+
+  trafo_rat = function(R) # transformes a point on E_int back into a point on E
+    if R.isinfinite
+      return infinity(E)
+    end
+
+    xnew = divexact(R.coordx, mue^2)
+    ynew = divexact(R.coordy, mue^3)
+    S = E([xnew, ynew])
+    return S
+  end
+
+  return E_int::EllCrv{fmpq}, trafo_int, trafo_rat
+end
+
+@doc Markdown.doc"""
+    integral_model(E::EllCrv{fmpq}) -> (F::EllCrv{nf_elem}, function, function)
```

Comment:
the input here is stated as fmpq, the text allows also nf_elem. Further are the 2nd and 3rd return value really julia functios or are they maps? Ie. can I ask for domain/ codomain....

Reply:
They are probably EllCrvIsos, which are Maps. I probably copy-pasted the old docstring for integral_model and forgot to change it.